File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/CourseReviewsDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * DTO for paginated course reviews response.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CourseReviewsDTO {
    private List<ReviewDTO> reviews;
    private Integer totalCount;
    private Integer currentPage;
    private Boolean hasNext;
    private Boolean hasPrevious;
    private String nextUrl;
    private String previousUrl;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/CourseReviewsDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for paginated course reviews response.
 */
data class CourseReviewsDTO(
    val reviews: List<ReviewDTO>? = null,
    val totalCount: Int? = null,
    val currentPage: Int? = null,
    val hasNext: Boolean? = null,
    val hasPrevious: Boolean? = null,
    val nextUrl: String? = null,
    val previousUrl: String? = null,
)
